File: app/src/main/java/ru/magflayer/spectrum/injection/InjectorManager.java
package ru.magflayer.spectrum.injection;

import android.app.Application;

public class InjectorManager {

    private static AppComponent appComponent;

    public static void init(Application application) {
        synchronized (InjectorManager.class) {
            appComponent = DaggerAppComponent.builder()
                    .appModule(new AppModule(application))
                    .build();
        }
    }

    public static AppComponent getAppComponent() {
        return appComponent;
    }
}

File: app/src/main/java/ru/magflayer/spectrum/domain/injection/InjectorManager.kt
package ru.magflayer.spectrum.domain.injection

import android.app.Application

object InjectorManager {

    var appComponent: AppComponent? = null
        private set

    fun init(application: Application) {
        synchronized(InjectorManager::class.java) {
            appComponent = DaggerAppComponent.builder()
                .appModule(AppModule(application))
                .build()
        }
    }
}
